lib/commands/intent.js

```diff
@@ -0,0 +1,274 @@
+  }
+  if (!_.isNil(display)) {
+    cmd.push('--display', display);
+  }
+  cmd.push(...(parseCommonIntentArguments(opts)));
+  cmd.push(intent);
+  return await this.adb.shell(cmd);
+};
+
+/**
+ * @typedef {Object} StartServiceOptions
+ * @property {!string} intent - The name of the service intent to start, for example
+ * `com.some.package.name/.YourServiceSubClassName`. This option is mandatory.
+ * @property {?string|number} user ['current'] - The user ID for which the service is started.
+ * The `current` user id is used by default
+ * @property {?boolean} foreground [false] - Set it to `true` if your service must be
+ * started as foreground service.
+ * @property {?string} action - Action name
+ * @property {?string} uri - Unified resource identifier
+ * @property {?string} mimeType - Mime type
+ * @property {?string} identifier - Optional identifier
+ * @property {?string|Array<string>} categories - One or more category names
+ * @property {?string} component - Component name
+ * @property {Array<string|Array<string>>} extras - Optional intent arguments. Must be represented
+ * as array of arrays, where each subarray item contains two items: value type and the value itself.
+ * Supported value types are:
+ * - s: string. Value must be a valid string
+ * - sn: null. Value is ignored for this type
+ * - z: boolean. Value must be either `true` or `false`
+ * - i: integer. Value must be a valid 4-byte integer number
+ * - l: long. Value must be a valid 8-byte long number
+ * - f: float: Value must be a valid float number
+ * - u: uri. Value must be a valid uniform resource identifier string
+ * - cn: component name. Value must be a valid component name string
+ * - ia: Integer[]. Value must be a string of comma-separated integers
+ * - ial: List<Integer>. Value must be a string of comma-separated integers
+ * - la: Long[]. Value must be a string of comma-separated long numbers
+ * - lal: List<Long>. Value must be a string of comma-separated long numbers
+ * - fa: Float[]. Value must be a string of comma-separated float numbers
+ * - fal: List<Float>. Value must be a string of comma-separated float numbers
+ * - sa: String[]. Value must be comma-separated strings. To embed a comma into a string,
+ * escape it using "\,"
+ * - sal: List<String>. Value must be comma-separated strings. To embed a comma into a string,
+ * escape it using "\,"
+ * For example: [['s', 'My String1'], ['s', 'My String2'], ['ia', '1,2,3,4']]
+ */
+
+/**
+ * Starts the given service intent.
+ *
+ * @param {StartServiceOptions} opts
+ * @returns {string} The command output
+ * @throws {Error} If there was a failure while starting the service
+ * or required options are missing
+ */
+commands.mobileStartService = async function mobileStartService (opts = {}) {
+  const {
+    intent,
+    user,
+    foreground,
+  } = requireOptions(opts, ['intent']);
+  const cmd = [
+    'am', foreground ? 'start-foreground-service' : 'start-service',
```

Comment:
It seems `start-service` should be `startservice` for lower than Android 8 as same as start-activity.

Reply:
fixed